src/main/groovy/com/avast/gradle/dockercompose/TasksConfigurator.groovy

```diff
@@ -122,15 +123,35 @@ class TasksConfigurator {
     void isRequiredByCore(Task task, boolean fromConfigure) {
         task.dependsOn upTask
         task.finalizedBy downTask
+        def taskDependencies = getTaskDependencies(task)
         if (fromConfigure) {
-            upTask.get().shouldRunAfter task.taskDependencies
+            upTask.get().shouldRunAfter taskDependencies
         } else {
-            upTask.configure { it.shouldRunAfter task.taskDependencies }
+            upTask.configure { it.shouldRunAfter taskDependencies }
```

Comment:
IMHO, correct me if I´m wrong, the `getTaskDependencies(task)` should be called from inside the `configure { }` block to ensure the dependencies get resolved at the right time.

Reply:
Maybe we wouldn't have to wrap the dependencies into `TaskDependency` then 🤞

Reply:
Hmm, that does appear to work (not break), but now that I'm looking closer:
* i'm not convinced it's behaving (I can see it composing up before the other tasks are done). I'm not sure if this is just gradle doing stuff in parallel
* it looks like we're adding a circular dependency on the up task (because it's a dependency of the test task)

I'll have to spend some more time on this....

Reply:
I suspect [this](https://docs.gradle.org/current/userguide/controlling_task_execution.html#sec:ordering_tasks) comes into play:

> The "should run after" ordering rule is similar but less strict, as it will be ignored in two situations:
> 
> If using that rule introduces an ordering cycle.
> 
> When using parallel execution and all task dependencies have been satisfied apart from the "should run after" task, then this task will be run regardless of whether or not its "should run after" dependencies have been run.
>

Reply:
Ok, I've addressed this comment. Everything is working on my end so I'm happy with this now.